zkvm/src/vm.rs

```diff
@@ -317,6 +317,13 @@ where
         Ok(())
     }
 
+    fn range(&mut self, i: u8) -> Result<(), VMError> {
+        let expr = self.pop_item()?.to_expression()?;
+        self.add_range_proof(i as usize, expr.clone());
```

Comment:
Let's make `add_range_proof` failable and check that `i` is between `1` and `64` (0 and >64 are not allowed).